src/lib.rs

```diff
@@ -0,0 +1,324 @@
+use crate::exec::{
+    is_addr_in_exec, is_common_parent, AtomicExec, AtomicExecParamsRaw, ExecStatus, LockedOutput,
+    SubmitExecParams, SubmitOutput,
+};
```

Comment:
[Idiomatic Rust](https://doc.rust-lang.org/book/ch07-04-bringing-paths-into-scope-with-the-use-keyword.html#creating-idiomatic-use-paths) avoids bringing functions from other modules into scope.
```suggestion
use crate::exec::{
    AtomicExec, AtomicExecParamsRaw, ExecStatus, LockedOutput,
    SubmitExecParams, SubmitOutput,
};
```

Reply:
this would break the build though?

Reply:
An idiomatic way would be to also import `crate::exec` itself and to invoke those functions as `exec::is_addr_in_exec` and `exec::is_common_parent`.